File: backend/src/main/java/com/learn/EmployeeManagementSystem/service/AttendanceService.java
package com.learn.EmployeeManagementSystem.service;

import com.learn.EmployeeManagementSystem.DTOs.requestDto.AttendanceRequest;
import com.learn.EmployeeManagementSystem.DTOs.responseDTO.AttendanceResponse;

import java.util.List;

public interface AttendanceService
{
    // Get all attendances
    List<AttendanceResponse> getAllAttendance();

    //Get attendance by Id
    AttendanceResponse getAttendanceById(Long id);

    //Create attendance
    AttendanceResponse createAttendance(AttendanceRequest attendanceRequest);

    //update attendance
    AttendanceResponse updateAttendance(Long id,AttendanceRequest attendanceRequest);

    //Delete attendance
    void deleteAttendance(Long id);

}

File: frontend/frontend/src/services/attendanceService.js
import api from './api';

export const attendanceService = {
  // GET /attendance
  getAllAttendance: async () => {
    const response = await api.get('/attendance');
    return response.data;
  },

  // GET /attendance/{id}
  getAttendanceById: async (id) => {
    const response = await api.get(`/attendance/${id}`);
    return response.data;
  },

  // POST /attendance
  createAttendance: async (attendanceData) => {
    const response = await api.post('/attendance', attendanceData);
    return response.data;
  },

  // PUT /attendance/{id}
  updateAttendance: async (id, attendanceData) => {
    const response = await api.put(`/attendance/${id}`, attendanceData);
    return response.data;
  },

  // DELETE /attendance/{id}
  deleteAttendance: async (id) => {
    const response = await api.delete(`/attendance/${id}`);
    return response.data;
  },
};

export default attendanceService;
